lib/components/Icon.js

```diff
@@ -1,24 +1,65 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 
-/**
- * Abstracts away the icon implementation.
- */
-const Icon = ({ name, className, title }) => (
-    <span className={`icon ${name} ${className}`} aria-hidden={!title}>
-        {title ? <span role="presentation">{title}</span> : null}
-    </span>
-);
-
-Icon.propTypes = {
-    name: PropTypes.string.isRequired,
-    className: PropTypes.string,
+// The icon definition is very flexible.
+export const iconPropType = PropTypes.oneOfType([
+    // String icon = SVG path (d attribute).
+    PropTypes.string,
+    // List of SVG paths.
+    PropTypes.arrayOf(PropTypes.string),
+    // Arbitrary React element.
+    PropTypes.node,
+]);
+
+const propTypes = {
+    icon: iconPropType.isRequired,
     title: PropTypes.string,
+    className: PropTypes.string,
 };
 
-Icon.defaultProps = {
-    className: '',
+const defaultProps = {
     title: null,
+    className: null,
 };
 
+/**
+ * Icon as SVG element. Can optionally render a React element instead.
+ */
+const Icon = ({ icon, title, className }) => {
+    const isPathOrRef = typeof icon === 'string';
+    const isPathArray = Array.isArray(icon);
+    let children;
+
+    if (isPathOrRef) {
+        if (icon.startsWith('#')) {
```

Comment:
Should be contains – may not be the first character if it's a reference to an external file.